### DISCUSSION TITLE
Is it good practise to use Prisma ORM (for PostgreSQL) in NextJS app or raw SQL query?

### DISCUSSION THREAD

**Discussion by cocoblock11** [vercel/next.js#53580] [Help] [answered]
### Summary

I am starting up with NextJS and I want to re-develop one application and I want use NextJS with PostgreSQL database. I found Prisma ORM. So my question is, is it best practise to use Prisma ORM in NextJS application for PostgreSQL? There are approx 50000 users and daily active users are approx 25000 to 30000.

Or is there anything else I can use with NextJS and PostgreSQL?

Thank you for your help.

**Comment by masterkain** [comment, marked as answer]
usually it makes sense but you have to keep in mind that prisma comes with some strings attached:

- data proxy/pgbouncer are mandatory
- many commands to learn and mess with your database
- migration running in prod before deployment (shouldn't be too bad)
- syntax change between versions
- memory leaks
- might bring unused dependencies in your project
- bigints creating issues all over the place (you aren't forced to use this datatype though)
- overall weird design decisions (won't go into details but you can take a look at their issue tracker)

good luck

**Reply by cocoblock11** [reply]
Thank you for you response. I am considering Drizzle. Prisma uses Rust in the middle and its heavy too. So I will try Drizzle for beta version and check the performance.

**Reply by BrunoPierca** [reply]
Did you proceed? How did it come out? Thx
